concordium-consensus/src/Concordium/GlobalState/TreeState.hs

```diff
@@ -71,6 +73,15 @@ data AddTransactionResult =
   ObsoleteNonce
   deriving(Eq, Show)
 
+-- |The Transaction verification cache stores a transaction `VerificationResult` alongside its
+-- `TransactionHash`.
+-- New entries are being put into the cache via the Updater.
+-- The cached verification results is then used by the Scheduler for allowing for an easier verification process during
+-- during block execution.
+-- Entries in the cache are being removed when the associated transaction is either
+-- finalized or purged.
+type TransactionVerificationCache = HM.HashMap TransactionHash TV.VerificationResult
```

Comment:
Maybe. But this would imply that the transaction table would be exposed for the Scheduler? I would presume that this is not desirable as execution of transactions only depends on the block state, and the scheduler shouldn't know anything of the transaction table?

Reply:
I put it into the TransactionVerification module.